projects/vdk-plugins/vdk-meta-jobs/src/vdk/plugin/meta_jobs/time_based_queue.py

```diff
@@ -0,0 +1,66 @@
+# Copyright 2021 VMware, Inc.
+# SPDX-License-Identifier: Apache-2.0
+import time
+from collections import deque
+from dataclasses import dataclass
+from random import randint
+from typing import Any
+
+
+@dataclass
+class _QueueEntry:
+    element: Any
+    ready_time: float
+
+
+class TimeBasedQueue:
+    """
+    Use it to enqueue elements until element is ready to be dequeued.
+    """
+
+    def __init__(
+        self,
+        min_ready_time_seconds=10,
+        randomize_delay_seconds=10,
```

Comment:
why is there some randomized delay added?

Reply:
TimeBasedQueue is used to enable delayed re-tries. And we want the delay to be randomized so that not all re-tries trigger in the same time.